packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -466,17 +466,70 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
         const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
 
+        let isOptionsCall = false;
+
+        if (headers) {
```

Comment:
Even if there are no headers, the object would be empty liek {}. So the check is wrong. You probably want to check the length of object keys. If that is not 0 then it would be options call.

Reply:
Also you can check in local host, if the value is actually set to what we want

Reply:
I added check to if (headers.Authorization) which will be null if it's not options call. Also, we can't check in local host as this is odsp driver code and localhost doesn't use this.